diff --git a/src/Makefile.shlib b/src/Makefile.shlib
index 92d893e..4801b7a 100644
--- a/src/Makefile.shlib
+++ b/src/Makefile.shlib
@@ -187,7 +187,13 @@ endif
 ifeq ($(PORTNAME), linux)
   LINK.shared		= $(COMPILER) -shared
   ifdef soname
-    LINK.shared		+= -Wl,-soname,$(soname)
+    ifeq (linux-android,$(findstring linux-android,$(host_os)))
+        #crosscompiling for android needs to create unvesioned libs
+        shlib		= lib$(NAME)$(DLSUFFIX)
+        LINK.shared		+= -Wl,-soname,lib$(NAME)$(DLSUFFIX)
+    else
+        LINK.shared		+= -Wl,-soname,$(soname)
+    endif
   endif
   BUILD.exports		= ( echo '{ global:'; $(AWK) '/^[^\#]/ {printf "%s;\n",$$1}' $<; echo ' local: *; };' ) >$@
   exports_file		= $(SHLIB_EXPORTS:%.txt=%.list)
