diff --git a/qwebdav.cpp b/qwebdav.cpp
index 28478cd..bd8c646 100644
--- a/qwebdav.cpp
+++ b/qwebdav.cpp
@@ -410,13 +410,12 @@ QNetworkReply* QWebdav::get(const QString& path)
 
     QUrl reqUrl(m_baseUrl);
     reqUrl.setPath(absolutePath(path));
+    req.setUrl(reqUrl);
 
 #ifdef DEBUG_WEBDAV
     qDebug() << "QWebdav::get() url = " << req.url().toString(QUrl::RemoveUserInfo);
 #endif
 
-    req.setUrl(reqUrl);
-
     return QNetworkAccessManager::get(req);
 }
 
diff --git a/qwebdavdirparser.cpp b/qwebdavdirparser.cpp
index a48f3cd..15cfc99 100644
--- a/qwebdavdirparser.cpp
+++ b/qwebdavdirparser.cpp
@@ -70,7 +70,7 @@ QWebdavDirParser::~QWebdavDirParser()
     }
 }
 
-bool QWebdavDirParser::listDirectory(QWebdav *pWebdav, const QString &path)
+bool QWebdavDirParser::listDirectory(QWebdav *pWebdav, const QString &path, bool recursive)
 {
     if (m_busy)
         return false;
@@ -93,7 +93,7 @@ bool QWebdavDirParser::listDirectory(QWebdav *pWebdav, const QString &path)
     m_abort = false;
     m_includeRequestedURI = false;
 
-    m_reply = pWebdav->list(path);
+    m_reply = pWebdav->list(path, recursive ? 2 : 1);
     connect(m_reply, SIGNAL(finished()), this, SLOT(replyFinished()));
 
     if (!m_dirList.isEmpty())
diff --git a/qwebdavdirparser.h b/qwebdavdirparser.h
index 2943e7f..cb1422b 100644
--- a/qwebdavdirparser.h
+++ b/qwebdavdirparser.h
@@ -73,7 +73,7 @@ public:
     ~QWebdavDirParser();
 
     //! get all items of a collection
-    bool listDirectory(QWebdav *pWebdav, const QString &path);
+    bool listDirectory(QWebdav *pWebdav, const QString &path, bool recursive = false);
     //! get only information about the collection
     bool getDirectoryInfo(QWebdav *pWebdav, const QString &path);
     //! get only information about a file
