diff --git a/CMakeLists.txt b/CMakeLists.txt
index b2b3fef..4302e3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,16 +4,16 @@ find_package(QT NAMES Qt6 Qt5 COMPONENTS Network Xml REQUIRED)
 find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Network Xml REQUIRED)
 set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR})
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR})
-add_library(QtWebDAV SHARED
-        QNaturalSort.cpp
-        QNaturalSort.h
-        QWebDAV.cpp
-        QWebDAV.h
-        QWebDAV_global.h
-        QWebDAVDirParser.cpp
-        QWebDAVDirParser.h
-        QWebDAVItem.cpp
-        QWebDAVItem.h
+add_library(QtWebDAV STATIC
+        qnaturalsort.cpp
+        qnaturalsort.h
+        qwebdav.cpp
+        qwebdav.h
+        qwebdav_global.h
+        qwebdavdirparser.cpp
+        qwebdavdirparser.h
+        qwebdavitem.cpp
+        qwebdavitem.h
         )
 
 target_link_libraries(QtWebDAV PUBLIC
@@ -28,3 +28,8 @@ option(BUILD_EXAMPLE "Build with example")
 if (BUILD_EXAMPLE)
     add_subdirectory(example)
 endif()
+
+install(TARGETS QtWebDAV LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+file(GLOB QTWEBDAV_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+install(FILES ${QTWEBDAV_HEADERS} DESTINATION "include")
