diff --git a/src/analysis/processing/qgsalgorithmpackage.cpp b/src/analysis/processing/qgsalgorithmpackage.cpp
index faeafe9d71..e6fa56bac0 100644
--- a/src/analysis/processing/qgsalgorithmpackage.cpp
+++ b/src/analysis/processing/qgsalgorithmpackage.cpp
@@ -21,6 +21,8 @@
 #include "qgsvectorlayer.h"
 #include "qgssettings.h"
 
+#include <QLocale>
+
 ///@cond PRIVATE
 
 QString QgsPackageAlgorithm::name() const
diff --git a/src/core/qgis.h b/src/core/qgis.h
index f642f4ba00..d75035291b 100644
--- a/src/core/qgis.h
+++ b/src/core/qgis.h
@@ -54,6 +54,8 @@ int QgisEvent = QEvent::User + 1;
 class CORE_EXPORT Qgis
 {
     Q_GADGET
+    Q_CLASSINFO( "RegisterEnumClassesUnscoped", "false" )
+
   public:
 
     /**
diff --git a/src/core/raster/qgsrasterattributetable.cpp b/src/core/raster/qgsrasterattributetable.cpp
index a6aa1bc10a..49bcd83d5e 100644
--- a/src/core/raster/qgsrasterattributetable.cpp
+++ b/src/core/raster/qgsrasterattributetable.cpp
@@ -13,6 +13,7 @@
  *   (at your option) any later version.                                   *
  *                                                                         *
  ***************************************************************************/
+
 #include "qgsrasterattributetable.h"
 #include "qgsvectorfilewriter.h"
 #include "qgsogrprovider.h"
@@ -22,6 +23,9 @@
 #include "qgssinglebandpseudocolorrenderer.h"
 #include "qgsrastershader.h"
 #include "qgsrastershaderfunction.h"
+
+#include <QLocale>
+
 #include <mutex>
 #include <cmath>
 
