diff --git a/src/core/mesh/qgstopologicalmesh.h b/src/core/mesh/qgstopologicalmesh.h
index fbceaf5176..4f1546c5d8 100644
--- a/src/core/mesh/qgstopologicalmesh.h
+++ b/src/core/mesh/qgstopologicalmesh.h
@@ -20,7 +20,7 @@
 
 #include "qgsmeshdataprovider.h"
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && defined(CORE_EXPORT)
 template CORE_EXPORT QVector<int> SIP_SKIP;
 template CORE_EXPORT QList<int> SIP_SKIP;
 template CORE_EXPORT QVector<QVector<int>> SIP_SKIP;
