From 559afdb4111a1bb5208d6c01fc9a3ef587f03808 Mon Sep 17 00:00:00 2001
From: Mathieu Pellerin <nirvn.asia@gmail.com>
Date: Sun, 23 Oct 2022 09:16:04 +0700
Subject: [PATCH] Fix QgsVariantUtils::isNull not building on some compilers

---
 src/core/qgsvariantutils.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/core/qgsvariantutils.cpp b/src/core/qgsvariantutils.cpp
index 36ed32eee3..4dd47a65ed 100644
--- a/src/core/qgsvariantutils.cpp
+++ b/src/core/qgsvariantutils.cpp
@@ -366,7 +366,6 @@ bool QgsVariantUtils::isNull( const QVariant &variant )
       }
       return false;
 
-    case QVariant::LastCoreType:
     case QVariant::Color:
     case QVariant::Font:
     case QVariant::Brush:
@@ -381,7 +380,6 @@ bool QgsVariantUtils::isNull( const QVariant &variant )
     case QVariant::Transform:
     case QVariant::Matrix4x4:
     case QVariant::SizePolicy:
-    case QVariant::LastGuiType:
       break;
 
     case QVariant::UserType:
-- 
2.34.1

