diff --git a/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_System.h b/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_System.h
index 468c4fcc8..fa865ef82 100644
--- a/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_System.h
+++ b/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_System.h
@@ -34,6 +34,8 @@ extern "C" {
 #include "SentryCrashMonitor.h"
 #import "SentryInternalDefines.h"
 
+#include <stdint.h>
+
 /** Access the Monitor API.
  */
 SentryCrashMonitorAPI *sentrycrashcm_system_getAPI(void);
