diff -ur a/Sources/SentryCrash/Recording/Tools/SentryCrashMachineContext.c b/Sources/SentryCrash/Recording/Tools/SentryCrashMachineContext.c
--- a/Sources/SentryCrash/Recording/Tools/SentryCrashMachineContext.c   2022-08-07 06:08:36.000000000 +0000
+++ b/Sources/SentryCrash/Recording/Tools/SentryCrashMachineContext.c   2022-08-07 06:08:00.000000000 +0000
@@ -37,6 +37,7 @@
 #include "SentryCrashLogger.h"
 
 #ifdef __arm64__
+#include <sys/_types/_ucontext64.h>
 #    define UC_MCONTEXT uc_mcontext64
 typedef ucontext64_t SignalUserContext;
 #else
