From 171161eaba53e21ada2411227135982e49c6caee Mon Sep 17 00:00:00 2001
From: Matthias Kuhn <matthias@opengis.ch>
Date: Wed, 29 Dec 2021 13:02:59 +0100
Subject: [PATCH] Remove locale

---
 src/iso19111/internal.cpp | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/src/iso19111/internal.cpp b/src/iso19111/internal.cpp
index 4810202d..69f60680 100644
--- a/src/iso19111/internal.cpp
+++ b/src/iso19111/internal.cpp
@@ -272,14 +272,7 @@ double c_locale_stod(const std::string &s) {
         }
     }
 
-    std::istringstream iss(s);
-    iss.imbue(std::locale::classic());
-    double d;
-    iss >> d;
-    if (!iss.eof() || iss.fail()) {
-        throw std::invalid_argument("non double value");
-    }
-    return d;
+    return std::stod(s);
 }
 
 // ---------------------------------------------------------------------------
-- 
2.31.1

