diff --git a/CMakeLists.txt b/CMakeLists.txt
index b48b2d09..4f0255d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,9 +71,7 @@ if(NOT ZXING_DEPENDENCIES IN_LIST ZXING_DEPENDENCIES_LIST)
 endif()
 
 if (NOT DEFINED CMAKE_CXX_STANDARD)
-    set (CMAKE_CXX_STANDARD 20)
-    # Allow the fallback to earlier versions if the compiler does not support it.
-    set(CMAKE_CXX_STANDARD_REQUIRED OFF)
+    set (CMAKE_CXX_STANDARD 17)
 endif()
 if (NOT DEFINED CMAKE_CXX_EXTENSIONS)
     set (CMAKE_CXX_EXTENSIONS OFF)
