diff --color -Naur dc25a70291-1f7ed7b5aa.clean/cmake/FindPostgres.cmake b/cmake/FindPostgres.cmake
--- dc25a70291-1f7ed7b5aa.clean/cmake/FindPostgres.cmake	2021-10-02 01:04:42.000000000 +0200
+++ b/cmake/FindPostgres.cmake	2021-10-18 10:44:16.028140987 +0200
@@ -14,6 +14,14 @@
 #    POSTGRES_INCLUDE_DIR
 #    POSTGRES_LIBRARY
 
+find_package(PostgreSQL)
+if(${PostgreSQL_FOUND})
+  set(POSTGRES_INCLUDE_DIR ${PostgreSQL_INCLUDE_DIRS})
+  set(POSTGRES_LIBRARY ${PostgreSQL_LIBRARIES})
+  set(POSTGRES_FOUND TRUE)
+  return()
+endif()
+
 IF(ANDROID)
   SET(POSTGRES_INCLUDE_DIR ${POSTGRES_INCLUDE_DIR} CACHE STRING INTERNAL)
   SET(POSTGRES_LIBRARY ${PG_TMP}/libpq.so CACHE STRING INTERNAL)
