diff --git a/src/core/raster/qgshuesaturationfilter.cpp b/src/core/raster/qgshuesaturationfilter.cpp
index b1309be4ab..7389e5c2f4 100644
--- a/src/core/raster/qgshuesaturationfilter.cpp
+++ b/src/core/raster/qgshuesaturationfilter.cpp
@@ -173,12 +173,6 @@ QgsRasterBlock *QgsHueSaturationFilter::block( int bandNo, QgsRectangle  const &
 
     if ( mInvertColors || alpha != 255 )
     {
-      if ( mInvertColors )
-      {
-        r = 255 - r;
-        g = 255 - g;
-        b = 255 - b;
-      }
       if ( alpha != 255 )
       {
         // Semi-transparent pixel. We need to adjust the colors since we are using Qgis::DataType::ARGB32_Premultiplied
@@ -188,6 +182,12 @@ QgsRasterBlock *QgsHueSaturationFilter::block( int bandNo, QgsRectangle  const &
         g /= alphaFactor;
         b /= alphaFactor;
       }
+      if ( mInvertColors )
+      {
+        r = 255 - r;
+        g = 255 - g;
+        b = 255 - b;
+      }
       myColor = QColor::fromRgb( r, g, b );
     }
 
