diff --git a/src/providers/wfs/qgscachedirectorymanager.cpp b/src/providers/wfs/qgscachedirectorymanager.cpp
index 2b4cd1739c..cb5fc5d28d 100644
--- a/src/providers/wfs/qgscachedirectorymanager.cpp
+++ b/src/providers/wfs/qgscachedirectorymanager.cpp
@@ -157,6 +157,9 @@ bool QgsCacheDirectoryManager::removeDir( const QString &dirName )
 std::unique_ptr<QSharedMemory> QgsCacheDirectoryManager::createAndAttachSHM()
 {
   std::unique_ptr<QSharedMemory> sharedMemory;
+// See https://wiki.qt.io/Qt_for_iOS_known_issues:
+// Crashes: Program received signal SIGSYS, Bad system call. 0x3b23bbe4 in semget ()
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0) || !defined(Q_OS_IOS)
   // For debug purpose. To test in the case where shared memory mechanism doesn't work
   if ( !getenv( "QGIS_USE_SHARED_MEMORY_KEEP_ALIVE" ) )
   {
@@ -177,6 +180,7 @@ std::unique_ptr<QSharedMemory> QgsCacheDirectoryManager::createAndAttachSHM()
       }
     }
   }
+#endif
   return nullptr;
 }
 
