diff -Naur 3e0eed1a92-3d69790817.clean/CMakeLists.txt 3e0eed1a92-3d69790817/CMakeLists.txt
--- 3e0eed1a92-3d69790817.clean/CMakeLists.txt	2022-10-10 16:06:38.882706700 +0200
+++ 3e0eed1a92-3d69790817/CMakeLists.txt	2022-10-10 16:33:16.211988400 +0200
@@ -109,6 +109,10 @@
  set(CMAKE_DEBUG_POSTFIX "_debug")
 endif()
 
+if (MSVC)
+ add_definitions(-DNOMINMAX)
+endif()
+
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
    if (CMAKE_SYSTEM_NAME MATCHES Linux)
      add_definitions (-D_DEFAULT_SOURCE)
diff -Naur 3e0eed1a92-3d69790817.clean/cmake/modules/QcaMacro.cmake 3e0eed1a92-3d69790817/cmake/modules/QcaMacro.cmake
--- 3e0eed1a92-3d69790817.clean/cmake/modules/QcaMacro.cmake	2022-09-21 22:42:24.000000000 +0200
+++ 3e0eed1a92-3d69790817/cmake/modules/QcaMacro.cmake	2022-10-10 16:21:04.700565100 +0200
@@ -66,8 +66,11 @@
 
 macro(install_pdb TARGET INSTALL_PATH)
   if(MSVC)
-    install(FILES $<TARGET_PDB_FILE:${TARGET}> DESTINATION ${INSTALL_PATH} CONFIGURATIONS Debug)
-    install(FILES $<TARGET_PDB_FILE:${TARGET}> DESTINATION ${INSTALL_PATH} CONFIGURATIONS RelWithDebInfo)
+    get_target_property(target_type ${TARGET} TYPE)
+    if (target_type STREQUAL SHARED_LIBRARY)
+      install(FILES $<TARGET_PDB_FILE:${TARGET}> DESTINATION ${INSTALL_PATH} CONFIGURATIONS Debug OPTIONAL)
+      install(FILES $<TARGET_PDB_FILE:${TARGET}> DESTINATION ${INSTALL_PATH} CONFIGURATIONS RelWithDebInfo OPTIONAL)
+    endif()
   endif()
 endmacro(install_pdb)
 
