diff --git a/src/qca_safeobj.h b/src/qca_safeobj.h
index 5c47adad..20570b27 100644
--- a/src/qca_safeobj.h
+++ b/src/qca_safeobj.h
@@ -57,7 +57,7 @@ public Q_SLOTS:
     }
 
 Q_SIGNALS:
-    void activated();
+    void activated(QSocketDescriptor socket, QSocketNotifier::Type type);
 
 private:
     QSocketNotifier *sn;
diff --git a/src/support/qpipe.cpp b/src/support/qpipe.cpp
index 122e96a8..2d9d8ab7 100644
--- a/src/support/qpipe.cpp
+++ b/src/support/qpipe.cpp
@@ -1075,8 +1075,10 @@ public Q_SLOTS:
 #endif
     }
 
-    void sn_read_activated()
+    void sn_read_activated(QSocketDescriptor socket, QSocketNotifier::Type type)
     {
+        Q_UNUSED(socket)
+        Q_UNUSED(type)
 #ifdef Q_OS_UNIX
         if (blockReadNotify)
             return;
