diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 47773219..0e2b90ac 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -52,7 +52,7 @@ SET( SOURCES
 
 IF (WIN32)
   SET( SOURCES ${SOURCES} qca_systemstore_win.cpp )
-elseif(APPLE AND NOT IOS)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set( SOURCES ${SOURCES} qca_systemstore_mac.cpp)
 else()
   SET( SOURCES ${SOURCES} qca_systemstore_flatfile.cpp )
