diff --git a/CMakeLists.txt b/CMakeLists.txt
index cb8430f3591..ee0788efe7c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -540,7 +540,11 @@ if(WITH_CORE)
   endif()
 
   # Password helper
-  find_package(QtKeychain REQUIRED)
+  if(BUILD_WITH_QT6)
+    find_package(Qt6Keychain CONFIG REQUIRED)
+  else()
+    find_package(Qt5Keychain CONFIG REQUIRED)
+  endif()
   # Master password hash and authentication encryption
   find_package(QCA REQUIRED)
   # Check for runtime dependency of qca-ossl plugin
diff --git a/cmake/FindQtKeychain.cmake b/cmake/FindQtKeychain.cmake
deleted file mode 100644
index a32ebff397a..00000000000
--- a/cmake/FindQtKeychain.cmake
+++ /dev/null
@@ -1,52 +0,0 @@
-# Find QtKeychain
-# ~~~~~~~~~~~~~~~
-# Copyright (c) 2016, Boundless Spatial
-# Author: Larry Shaffer <lshaffer (at) boundlessgeo (dot) com>
-#
-# Redistribution and use is allowed according to the terms of the BSD license.
-# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
-#
-# CMake module to search for QtKeychain library from:
-#    https://github.com/frankosterfeld/qtkeychain
-#
-# If it's found it sets QTKEYCHAIN_FOUND to TRUE
-# and following variables are set:
-#    QTKEYCHAIN_INCLUDE_DIR
-#    QTKEYCHAIN_LIBRARY
-
-FIND_PATH(QTKEYCHAIN_INCLUDE_DIR keychain.h
-  PATHS
-  ${LIB_DIR}/include
-  "$ENV{LIB_DIR}/include"
-  $ENV{INCLUDE}
-  /usr/local/include
-  /usr/include
-  PATH_SUFFIXES ${QT_VERSION_BASE_LOWER}keychain qtkeychain
-)
-
-FIND_LIBRARY(QTKEYCHAIN_LIBRARY NAMES ${QT_VERSION_BASE_LOWER}keychain qtkeychain
-  PATHS
-  ${LIB_DIR}
-  "$ENV{LIB_DIR}"
-  $ENV{LIB_DIR}/lib
-  $ENV{LIB}
-  /usr/local/lib
-  /usr/lib
-)
-
-
-IF (QTKEYCHAIN_INCLUDE_DIR AND QTKEYCHAIN_LIBRARY)
-  SET(QTKEYCHAIN_FOUND TRUE)
-ELSE()
-  SET(QTKEYCHAIN_FOUND FALSE)
-ENDIF (QTKEYCHAIN_INCLUDE_DIR AND QTKEYCHAIN_LIBRARY)
-
-IF (QTKEYCHAIN_FOUND)
-   IF (NOT QTKEYCHAIN_FIND_QUIETLY)
-      MESSAGE(STATUS "Found QtKeychain: ${QTKEYCHAIN_LIBRARY}")
-   ENDIF (NOT QTKEYCHAIN_FIND_QUIETLY)
-ELSE (QTKEYCHAIN_FOUND)
-   IF (QTKEYCHAIN_FIND_REQUIRED)
-      MESSAGE(FATAL_ERROR "Could not find QtKeychain")
-   ENDIF (QTKEYCHAIN_FIND_REQUIRED)
-ENDIF (QTKEYCHAIN_FOUND)
diff --git a/external/o2/src/o0keychainstore.cpp b/external/o2/src/o0keychainstore.cpp
index 2fe1d0e1ea9..d22591dce44 100644
--- a/external/o2/src/o0keychainstore.cpp
+++ b/external/o2/src/o0keychainstore.cpp
@@ -4,7 +4,11 @@
 #include "o0keychainstore.h"
 
 #include <QDebug>
-#include <keychain.h>
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+#include <qt6keychain/keychain.h>
+#else
+#include <qt5keychain/keychain.h>
+#endif
 #include <QtCore/QDataStream>
 #include <QtCore/QBuffer>
 #include <QtCore/QEventLoop>
diff --git a/src/auth/oauth2/CMakeLists.txt b/src/auth/oauth2/CMakeLists.txt
index 6bb0c665327..a9e2cfeddc3 100644
--- a/src/auth/oauth2/CMakeLists.txt
+++ b/src/auth/oauth2/CMakeLists.txt
@@ -3,9 +3,6 @@
 ########################################################
 # Packages
 
-if(NOT QTKEYCHAIN_FOUND)
-  find_package(QtKeychain REQUIRED)
-endif()
 option(WITH_INTERNAL_O2 "Download and locally include source of o2 library" ON)
 
 if(WITH_INTERNAL_O2)
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index a802a8f6431..56f9c9e96aa 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -2416,6 +2416,16 @@ target_link_libraries(qgis_core
   PROJ::proj
 )
 
+if(BUILD_WITH_QT6)
+target_link_libraries(qgis_core
+  Qt6Keychain::Qt6Keychain
+)
+else()
+target_link_libraries(qgis_core
+  Qt5Keychain::Qt5Keychain
+)
+endif()
+
 if (WITH_DRACO)
   target_link_libraries(qgis_core ${DRACO_LIBRARY})
 endif()
diff --git a/src/core/auth/qgsauthmanager.cpp b/src/core/auth/qgsauthmanager.cpp
index 5948df209df..429c3e58b42 100644
--- a/src/core/auth/qgsauthmanager.cpp
+++ b/src/core/auth/qgsauthmanager.cpp
@@ -47,9 +47,6 @@
 #include <QSslConfiguration>
 #endif
 
-// QtKeyChain library
-#include "keychain.h"
-
 // QGIS includes
 #include "qgsauthcertutils.h"
 #include "qgsauthcrypto.h"
diff --git a/src/core/auth/qgsauthmanager.h b/src/core/auth/qgsauthmanager.h
index 6ab376d35ba..a29cc5b74ec 100644
--- a/src/core/auth/qgsauthmanager.h
+++ b/src/core/auth/qgsauthmanager.h
@@ -38,8 +38,11 @@
 #include "qgsauthconfig.h"
 #include "qgsauthmethod.h"
 
-// Qt5KeyChain library
-#include "keychain.h"
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+#include <qt6keychain/keychain.h>
+#else
+#include <qt5keychain/keychain.h>
+#endif
 
 #ifndef SIP_RUN
 namespace QCA
