diff --color -Naur b/CMakeLists.txt QGIS-final-3_28_0/CMakeLists.txt
--- b/CMakeLists.txt	2022-10-22 19:08:22.563229185 +0200
+++ QGIS-final-3_28_0/CMakeLists.txt	2022-10-22 19:11:24.686948005 +0200
@@ -493,11 +493,11 @@
   else()
     message(STATUS "Found Qt version: ${Qt5Core_VERSION_STRING}")
   endif()
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Android" AND NOT BUILD_WITH_QT6)
+    find_package(${QT_VERSION_BASE} COMPONENTS AndroidExtras)
+  endif()
   if (WITH_QUICK)
     find_package(${QT_VERSION_BASE} COMPONENTS Qml Quick REQUIRED)
-    if(${CMAKE_SYSTEM_NAME} MATCHES "Android" AND NOT BUILD_WITH_QT6)
-      find_package(${QT_VERSION_BASE} COMPONENTS AndroidExtras)
-    endif()
 
     # following variable is used in qgsconfig.h
     set (HAVE_QUICK TRUE)
