diff --git a/src/analysis/processing/qgsalgorithmaddincrementalfield.h b/src/analysis/processing/qgsalgorithmaddincrementalfield.h
index 237cfaa90b0..bfdbc7537a8 100644
--- a/src/analysis/processing/qgsalgorithmaddincrementalfield.h
+++ b/src/analysis/processing/qgsalgorithmaddincrementalfield.h
@@ -42,6 +42,7 @@ class QgsAddIncrementalFieldAlgorithm : public QgsProcessingFeatureBasedAlgorith
     QString shortHelpString() const override;
     QList<int> inputLayerTypes() const override;
     QgsAddIncrementalFieldAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -53,7 +54,6 @@ class QgsAddIncrementalFieldAlgorithm : public QgsProcessingFeatureBasedAlgorith
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureRequest request() const override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
 
diff --git a/src/analysis/processing/qgsalgorithmaddtablefield.h b/src/analysis/processing/qgsalgorithmaddtablefield.h
index 21ad7783c52..a3ea8370996 100644
--- a/src/analysis/processing/qgsalgorithmaddtablefield.h
+++ b/src/analysis/processing/qgsalgorithmaddtablefield.h
@@ -42,6 +42,7 @@ class QgsAddTableFieldAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString shortHelpString() const override;
     QList<int> inputLayerTypes() const override;
     QgsAddTableFieldAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -52,7 +53,6 @@ class QgsAddTableFieldAlgorithm : public QgsProcessingFeatureBasedAlgorithm
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
 
diff --git a/src/analysis/processing/qgsalgorithmaddxyfields.h b/src/analysis/processing/qgsalgorithmaddxyfields.h
index dc4466fe6f0..023280e04d0 100644
--- a/src/analysis/processing/qgsalgorithmaddxyfields.h
+++ b/src/analysis/processing/qgsalgorithmaddxyfields.h
@@ -43,6 +43,7 @@ class QgsAddXYFieldsAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString shortDescription() const override;
     QList<int> inputLayerTypes() const override;
     QgsAddXYFieldsAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -54,7 +55,6 @@ class QgsAddXYFieldsAlgorithm : public QgsProcessingFeatureBasedAlgorithm
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
 
diff --git a/src/analysis/processing/qgsalgorithmaggregate.h b/src/analysis/processing/qgsalgorithmaggregate.h
index cb6709e8f70..14fa03602cb 100644
--- a/src/analysis/processing/qgsalgorithmaggregate.h
+++ b/src/analysis/processing/qgsalgorithmaggregate.h
@@ -41,6 +41,7 @@ class QgsAggregateAlgorithm : public QgsProcessingAlgorithm
     QString group() const override;
     QString groupId() const override;
     QString shortHelpString() const override;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
     QgsAggregateAlgorithm *createInstance() const override SIP_FACTORY;
     void initAlgorithm( const QVariantMap &configuration = QVariantMap() ) override;
 
@@ -49,8 +50,6 @@ class QgsAggregateAlgorithm : public QgsProcessingAlgorithm
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QVariantMap processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
 
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
-
   private:
 
     QgsExpression createExpression( const QString &expressionString, QgsProcessingContext &context ) const;
diff --git a/src/analysis/processing/qgsalgorithmdropfields.h b/src/analysis/processing/qgsalgorithmdropfields.h
index 7a2e27e4eef..1b8053002b3 100644
--- a/src/analysis/processing/qgsalgorithmdropfields.h
+++ b/src/analysis/processing/qgsalgorithmdropfields.h
@@ -44,6 +44,7 @@ class QgsDropTableFieldsAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString shortDescription() const override;
     QList<int> inputLayerTypes() const override;
     QgsDropTableFieldsAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -54,7 +55,6 @@ class QgsDropTableFieldsAlgorithm : public QgsProcessingFeatureBasedAlgorithm
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
 
diff --git a/src/analysis/processing/qgsalgorithmextractzmvalues.h b/src/analysis/processing/qgsalgorithmextractzmvalues.h
index aa2a7ab969a..80ffdaf1c94 100644
--- a/src/analysis/processing/qgsalgorithmextractzmvalues.h
+++ b/src/analysis/processing/qgsalgorithmextractzmvalues.h
@@ -39,6 +39,7 @@ class QgsExtractZMValuesAlgorithmBase : public QgsProcessingFeatureBasedAlgorith
     QString group() const override;
     QString groupId() const override;
     QList<int> inputLayerTypes() const override;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -49,7 +50,6 @@ class QgsExtractZMValuesAlgorithmBase : public QgsProcessingFeatureBasedAlgorith
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
diff --git a/src/analysis/processing/qgsalgorithmfieldcalculator.h b/src/analysis/processing/qgsalgorithmfieldcalculator.h
index ebd375713fd..38b3f9685a2 100644
--- a/src/analysis/processing/qgsalgorithmfieldcalculator.h
+++ b/src/analysis/processing/qgsalgorithmfieldcalculator.h
@@ -44,6 +44,7 @@ class QgsFieldCalculatorAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString shortHelpString() const override;
     QList<int> inputLayerTypes() const override;
     QgsFieldCalculatorAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
     void initParameters( const QVariantMap &configuration = QVariantMap() ) override;
@@ -53,7 +54,6 @@ class QgsFieldCalculatorAlgorithm : public QgsProcessingFeatureBasedAlgorithm
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
     QgsFields mFields;
diff --git a/src/analysis/processing/qgsalgorithmpointsinpolygon.h b/src/analysis/processing/qgsalgorithmpointsinpolygon.h
index aa07f5891c6..66cdd0c314d 100644
--- a/src/analysis/processing/qgsalgorithmpointsinpolygon.h
+++ b/src/analysis/processing/qgsalgorithmpointsinpolygon.h
@@ -48,6 +48,7 @@ class QgsPointsInPolygonAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QList<int> inputLayerTypes() const override;
     Qgis::ProcessingSourceType outputLayerType() const override;
     QgsCoordinateReferenceSystem outputCrs( const QgsCoordinateReferenceSystem &inputCrs ) const override;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
 
   protected:
@@ -57,7 +58,6 @@ class QgsPointsInPolygonAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFields outputFields( const QgsFields &inputFields ) const override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
     bool mIsInPlace = false;
diff --git a/src/analysis/processing/qgsalgorithmrefactorfields.h b/src/analysis/processing/qgsalgorithmrefactorfields.h
index 7982777e747..9921d04d65c 100644
--- a/src/analysis/processing/qgsalgorithmrefactorfields.h
+++ b/src/analysis/processing/qgsalgorithmrefactorfields.h
@@ -43,6 +43,7 @@ class QgsRefactorFieldsAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString shortHelpString() const override;
     QList<int> inputLayerTypes() const override;
     QgsRefactorFieldsAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -53,7 +54,6 @@ class QgsRefactorFieldsAlgorithm : public QgsProcessingFeatureBasedAlgorithm
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
 
diff --git a/src/analysis/processing/qgsalgorithmrenametablefield.h b/src/analysis/processing/qgsalgorithmrenametablefield.h
index 5f4d159937c..14c4e9c97d7 100644
--- a/src/analysis/processing/qgsalgorithmrenametablefield.h
+++ b/src/analysis/processing/qgsalgorithmrenametablefield.h
@@ -43,6 +43,7 @@ class QgsRenameTableFieldAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString shortDescription() const override;
     QList<int> inputLayerTypes() const override;
     QgsRenameTableFieldAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
 
@@ -53,7 +54,6 @@ class QgsRenameTableFieldAlgorithm : public QgsProcessingFeatureBasedAlgorithm
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
 
diff --git a/src/analysis/processing/qgsalgorithmsetmvalue.h b/src/analysis/processing/qgsalgorithmsetmvalue.h
index 4879e2793c6..0331c8c9a6c 100644
--- a/src/analysis/processing/qgsalgorithmsetmvalue.h
+++ b/src/analysis/processing/qgsalgorithmsetmvalue.h
@@ -41,6 +41,7 @@ class QgsSetMValueAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString groupId() const override;
     QString shortHelpString() const override;
     QgsSetMValueAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *l ) const override;
 
   protected:
 
@@ -48,7 +49,6 @@ class QgsSetMValueAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString outputName() const override;
     Qgis::WkbType outputWkbType( Qgis::WkbType inputWkbType ) const override;
     Qgis::ProcessingFeatureSourceFlags sourceFlags() const override;
-    bool supportInPlaceEdit( const QgsMapLayer *l ) const override;
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
diff --git a/src/analysis/processing/qgsalgorithmsetzvalue.h b/src/analysis/processing/qgsalgorithmsetzvalue.h
index 3333a117663..56e1f7e6841 100644
--- a/src/analysis/processing/qgsalgorithmsetzvalue.h
+++ b/src/analysis/processing/qgsalgorithmsetzvalue.h
@@ -41,6 +41,7 @@ class QgsSetZValueAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString groupId() const override;
     QString shortHelpString() const override;
     QgsSetZValueAlgorithm *createInstance() const override SIP_FACTORY;
+    bool supportInPlaceEdit( const QgsMapLayer *l ) const override;
 
   protected:
 
@@ -48,7 +49,6 @@ class QgsSetZValueAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QString outputName() const override;
     Qgis::WkbType outputWkbType( Qgis::WkbType inputWkbType ) const override;
     Qgis::ProcessingFeatureSourceFlags sourceFlags() const override;
-    bool supportInPlaceEdit( const QgsMapLayer *l ) const override;
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
diff --git a/src/analysis/processing/qgsalgorithmsumlinelength.h b/src/analysis/processing/qgsalgorithmsumlinelength.h
index bb9f80e4465..f53f4c58b03 100644
--- a/src/analysis/processing/qgsalgorithmsumlinelength.h
+++ b/src/analysis/processing/qgsalgorithmsumlinelength.h
@@ -47,7 +47,7 @@ class QgsSumLineLengthAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     QList<int> inputLayerTypes() const override;
     Qgis::ProcessingSourceType outputLayerType() const override;
     QgsCoordinateReferenceSystem outputCrs( const QgsCoordinateReferenceSystem &inputCrs ) const override;
-
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   protected:
     QString inputParameterName() const override;
@@ -56,7 +56,6 @@ class QgsSumLineLengthAlgorithm : public QgsProcessingFeatureBasedAlgorithm
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFields outputFields( const QgsFields &inputFields ) const override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
     bool mIsInPlace = false;
diff --git a/src/analysis/processing/qgsalgorithmzonalstatisticsfeaturebased.h b/src/analysis/processing/qgsalgorithmzonalstatisticsfeaturebased.h
index c2f5b23c8f1..192455ac121 100644
--- a/src/analysis/processing/qgsalgorithmzonalstatisticsfeaturebased.h
+++ b/src/analysis/processing/qgsalgorithmzonalstatisticsfeaturebased.h
@@ -41,7 +41,7 @@ class QgsZonalStatisticsFeatureBasedAlgorithm : public QgsProcessingFeatureBased
     QString groupId() const override;
     QString shortHelpString() const override;
     QList<int> inputLayerTypes() const override;
-
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
     QgsZonalStatisticsFeatureBasedAlgorithm *createInstance() const override SIP_FACTORY;
 
   protected:
@@ -52,7 +52,6 @@ class QgsZonalStatisticsFeatureBasedAlgorithm : public QgsProcessingFeatureBased
 
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
     QgsFeatureList processFeature( const QgsFeature &feature,  QgsProcessingContext &context, QgsProcessingFeedback *feedback ) override;
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
   private:
     std::unique_ptr< QgsRasterInterface > mRaster;
diff --git a/src/core/processing/qgsprocessingalgorithm.h b/src/core/processing/qgsprocessingalgorithm.h
index 7191f809540..77fc871d251 100644
--- a/src/core/processing/qgsprocessingalgorithm.h
+++ b/src/core/processing/qgsprocessingalgorithm.h
@@ -461,6 +461,15 @@ class CORE_EXPORT QgsProcessingAlgorithm
      */
     void setProvider( QgsProcessingProvider *provider ) SIP_HOLDGIL;
 
+    /**
+     * Checks whether this algorithm supports in-place editing on the given \a layer
+     * Default implementation returns FALSE.
+     *
+     * \return TRUE if the algorithm supports in-place editing
+     * \since QGIS 3.4
+     */
+    virtual bool supportInPlaceEdit( const QgsMapLayer *layer ) const;
+
   protected:
 
     /**
@@ -1082,15 +1091,6 @@ class CORE_EXPORT QgsProcessingAlgorithm
      */
     static QString writeFeatureError( QgsFeatureSink *sink, const QVariantMap &parameters, const QString &name );
 
-    /**
-     * Checks whether this algorithm supports in-place editing on the given \a layer
-     * Default implementation returns FALSE.
-     *
-     * \return TRUE if the algorithm supports in-place editing
-     * \since QGIS 3.4
-     */
-    virtual bool supportInPlaceEdit( const QgsMapLayer *layer ) const;
-
   private:
 
     QgsProcessingProvider *mProvider = nullptr;
@@ -1175,9 +1175,15 @@ class CORE_EXPORT QgsProcessingFeatureBasedAlgorithm : public QgsProcessingAlgor
      */
     virtual QgsFeatureList processFeature( const QgsFeature &feature, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) SIP_THROW( QgsProcessingException ) = 0 SIP_VIRTUALERRORHANDLER( processing_exception_handler );
 
-  protected:
-
-    void initAlgorithm( const QVariantMap &configuration = QVariantMap() ) override;
+    /**
+     * Checks whether this algorithm supports in-place editing on the given \a layer
+     * Default implementation for feature based algorithms run some basic compatibility
+     * checks based on the geometry type of the layer.
+     *
+     * \return TRUE if the algorithm supports in-place editing
+     * \since QGIS 3.4
+     */
+    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
 
     /**
      * Returns the name of the parameter corresponding to the input layer.
@@ -1197,6 +1203,10 @@ class CORE_EXPORT QgsProcessingFeatureBasedAlgorithm : public QgsProcessingAlgor
      */
     virtual QString inputParameterDescription() const SIP_HOLDGIL;
 
+  protected:
+
+    void initAlgorithm( const QVariantMap &configuration = QVariantMap() ) override;
+
     /**
      * Returns the translated, user visible name for any layers created by this algorithm.
      * This name will be used as the default name when loading the resultant layer into a
@@ -1282,16 +1292,6 @@ class CORE_EXPORT QgsProcessingFeatureBasedAlgorithm : public QgsProcessingAlgor
      */
     virtual QgsFeatureRequest request() const;
 
-    /**
-     * Checks whether this algorithm supports in-place editing on the given \a layer
-     * Default implementation for feature based algorithms run some basic compatibility
-     * checks based on the geometry type of the layer.
-     *
-     * \return TRUE if the algorithm supports in-place editing
-     * \since QGIS 3.4
-     */
-    bool supportInPlaceEdit( const QgsMapLayer *layer ) const override;
-
     /**
      * Read the source from \a parameters and \a context and set it
      *
